### PR TITLE
generate missing metadata and met.json

### PR DESCRIPTION
**Pull request by 928PJY**

### CODE REVIEW

File: src/docfx/build/legacy/files/LegacyMarkdown.cs [hunk 1]
```diff
@@ -30,29 +49,65 @@ internal static class LegacyMarkdown
...
+            var rawMetadata = new JObject(pageModel.Metadata.Metadata);
+            rawMetadata["toc_rel"] = pageModel.TocRelativePath;
+            rawMetadata["locale"] = pageModel.Locale;
+            rawMetadata["word_count"] = pageModel.WordCount;
+            rawMetadata["depot_name"] = docset.Config.Name;
+            rawMetadata["site_name"] = "Docs";
+            rawMetadata["version"] = 0;
+            rawMetadata["_op_rawTitle"] = $"<h1>{HttpUtility.HtmlEncode(pageModel.Metadata.Title)}</h1>";
+
+            rawMetadata["_op_canonicalUrlPrefix"] = $"https://{docset.Config.HostName}/{docset.Config.Locale}/{docset.Config.SiteBasePath}/";
```
Comment (yufeih) [inline src/docfx/build/legacy/files/LegacyMarkdown.cs]: url base?

File: src/docfx/build/legacy/files/LegacyMarkdown.cs [hunk 2]
```diff
@@ -30,29 +49,65 @@ internal static class LegacyMarkdown
...
 
-            legacyPageModel.RawMetadata.Metadata["depot_name"] = docset.Config.Name;
-            legacyPageModel.RawMetadata.Metadata["is_dynamic_rendering"] = true;
-            legacyPageModel.RawMetadata.Metadata["layout"] = docset.Config.GlobalMetadata.TryGetValue("layout", out JToken layout) ? (string)layout : "Conceptual";
+        private static string GenerateLegacyPageMetadata(JObject rawMetadata)
+        {
+            string pageMetadataOutput = string.Empty;
+            foreach (string item in pageMetadataOutputItems)
+            {
+                if (rawMetadata.TryGetValue(item, out JToken value))
```
Comment (yufeih) [inline src/docfx/build/legacy/files/LegacyMarkdown.cs]: what if this is an array?

File: src/docfx/build/legacy/files/LegacyMarkdown.cs [hunk 3]
```diff
@@ -30,29 +49,65 @@ internal static class LegacyMarkdown
...
+                {
+                    pageMetadataOutput += $"<meta name=\"{item}\" content=\"{(string)value}\" />" + System.Environment.NewLine;
```
Comment (yufeih) [inline src/docfx/build/legacy/files/LegacyMarkdown.cs]: handle html attribute escape using `HtmlUtility`

File: src/docfx/build/legacy/files/LegacyMarkdown.cs [hunk 4]
```diff
@@ -30,29 +49,65 @@ internal static class LegacyMarkdown
...
```
Comment (OsmondJiang) [inline src/docfx/build/legacy/files/LegacyMarkdown.cs]: >pageMetadataOutput [](start = 20, length = 18)

string builder?

File: src/docfx/config/Config.cs [hunk 5]
```diff
@@ -53,7 +53,7 @@ internal class Config
         /// <summary>
         /// The hostname
         /// </summary>
-        public readonly string HostName = "docs.microsoft.com";
+        public readonly string BaseUrl = "https://docs.microsoft.com";
```
Comment (superyyrrzz) [inline src/docfx/config/Config.cs]: >https://docs.microsoft.com [](start = 42, length = 26)

string.Empty?

File: src/docfx/build/legacy/files/LegacyMarkdown.cs [hunk 6]
```diff
@@ -10,13 +11,32 @@ namespace Microsoft.Docs.Build
 {
     internal static class LegacyMarkdown
     {
+        private static readonly string[] pageMetadataOutputItems =
```
Comment (superyyrrzz) [inline src/docfx/build/legacy/files/LegacyMarkdown.cs]: >pageMetadataOutputItems [](start = 41, length = 23)

prepend s_

File: src/docfx/build/legacy/files/LegacyMarkdown.cs [hunk 7]
```diff
@@ -10,13 +11,32 @@ namespace Microsoft.Docs.Build
...
+        {
+            "author", "breadcrumb_path", "depot_name", "description", "document_id",
+            "document_version_independent_id", "gitcommit", "keywords",
+            "locale", "ms.author", "ms.date", "ms.prod", "ms.topic", "original_content_git_url",
+            "page_type", "search.ms_docsetname", "search.ms_product", "search.ms_sitename", "site_name",
+            "toc_rel", "uhfHeaderId", "updated_at", "version", "word_count",
+        };
+
+        private static readonly string[] metadataOutputItems =
```
Comment (superyyrrzz) [inline src/docfx/build/legacy/files/LegacyMarkdown.cs]: >metadataOutputItems [](start = 41, length = 19)

prepend s_

**Pull request merged**